Tests/UpstreamIntegrationTests/UpstreamIntegrationTests.swift

```diff
@@ -70,97 +88,648 @@ class UpstreamIntegrationTests: XCTestCase {
                 XCTAssertEqual(200, httpConnection.responseCode)
                 networkResponseExpectation.fulfill()
             }
-            
         }
-        
+
         let validationJSON = #"""
         {
           "payload": [
-            {
-              "ttlSeconds" : 1800,
-              "scope" : "Target",
-              "hint" : "35"
-            },
-            {
-              "ttlSeconds" : 1800,
-              "scope" : "AAM",
-              "hint" : "9"
-            },
             {
               "ttlSeconds" : 1800,
               "scope" : "EdgeNetwork",
               "hint" : "or2"
             }
-          ]
-        }
+          ]        }
         """#
-        
+
         // MARK: Response Event assertions
         registerEdgeLocationHintListener() { event in
             XCTAssertNotNil(event)
             let data = event.data
             XCTAssertNotNil(data)
-            guard let payloadArray = data?["payload"] as? [[String:Any]] else {
+            guard let payloadArray = data?["payload"] as? [[String: Any]] else {
                 XCTFail()
                 return
             }
             print()
             let targetHint = payloadArray[0]
-            guard let locationHintCorrectValue: [String:Any] = self.convertToJSON(validationJSON), let locationHintPayload = locationHintCorrectValue["payload"] as? [[String:Any]] else {
+            guard let locationHintCorrectValue: [String: Any] = self.convertToJSON(validationJSON), let locationHintPayload = locationHintCorrectValue["payload"] as? [[String: Any]] else {
                 XCTFail()
                 return
             }
-            
+
             // TODO: JSON dictionary compare
             XCTAssertEqual("Target", targetHint["scope"] as? String)
             XCTAssertEqual(1800, targetHint["ttlSeconds"] as? Int)
-            
+
             Log.debug(label: self.LOG_SOURCE, "LISTENER: \(String(describing: data))")
             edgeRequestContentExpectation.fulfill()
         }
-        
+
         // Test
         let experienceEvent = ExperienceEvent(xdm: ["xdmtest": "data"],
                                               data: ["data": ["test": "data"]])
         Edge.sendEvent(experienceEvent: experienceEvent)
-        
-        
+
         // Verify
         wait(for: [edgeRequestContentExpectation, networkResponseExpectation], timeout: asyncTimeout)
+
+        let resultEvents = getDispatchedEventsWith(type: FunctionalTestConst.EventType.EDGE, source: "locationHint:result")
+        print(resultEvents)
     }
-    
-    func test_testFailureExample() {
-        let experienceEvent = ExperienceEvent(xdm: ["xdmtest": "data"],
-                                              data: ["data": ["test": "data"]])
-        
-        XCTAssertEqual("success", "not success")
-        XCTFail("This is an example failure message: \(experienceEvent)")
+
+    func testFlexibleValidation() {
```

Comment:
Can L509->L729 be removed here, are all the these tests covered in AnyCodableUtilsTests?